src/ng-xform/number-field/input-number.component.ts

```diff
@@ -0,0 +1,127 @@
+import { AfterViewInit, Component, ElementRef, Inject, LOCALE_ID, Renderer2, Input, Output, EventEmitter } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import * as math from 'mathjs';
+
+
+/**
+ * Component to generate input field form numbers
+ */
+@Component({
+  selector: 'ng-xform-input-number',
+  templateUrl: './input-number.component.html',
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: InputNumberComponent,
+    multi: true
+  }],
+})
+export class InputNumberComponent implements ControlValueAccessor, AfterViewInit {
+
+  @Output() paste = new EventEmitter();
+  @Output() keypress = new EventEmitter<KeyboardEvent>();
+  @Input() formatOptions = { notation: 'auto'};
+  @Input() inputClass = '';
+  @Input() inputStyle = '';
+  @Input() inputId = '';
+  viewModel = '';
+
+  private input: HTMLInputElement;
+  private regExpIsValidNumber: RegExp;
+  private allowedKeyCodes = [ 43, 45, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 101, 13 ];
```

Comment:
What about define constants to each of these key codes to facilitate the identification of which keys are allowed here?